### PR TITLE
fix(benchmarks): update everything

### PR DESCRIPTION
**Pull request by dan-online**

### CODE REVIEW

File: benchmarks/src/lib/Benchmark.ts [hunk 1]
```diff
@@ -43,10 +42,10 @@ export class Benchmark {
     for (const [name, provider] of this.providers) {
       console.log(`${blueBright('Benchmark:')} ${cyanBright(name)}\n`);
 
-      const josh = new Josh<Benchmark.TestCard>({ name: 'benchmark', provider });
+      const josh = provider;
 
-      await josh.init();
-      await josh.clear();
+      await josh.init({ name: 'benchmark' });
+      await josh.clear({ method: Method.Clear, errors: [] });
```
Comment (RealShadowNova) [inline benchmarks/src/lib/Benchmark.ts:48]:
```suggestion
      await josh[Method.Clear]({ method: Method.Clear, errors: [] });
```

File: benchmarks/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "benchmarks",
+  "private": true,
+  "version": "1.0.0",
+  "description": "Benchmark for @joshdb providers",
+  "author": "@joshdb",
+  "dependencies": {
+    "@joshdb/provider": "2.0.0-next.6cc7d77.0"
+  }
+}
```
Comment (RealShadowNova) [inline benchmarks/package.json:10]: Could probably add scripts here for the benchmarks to run.

File: benchmarks/src/lib/constants/benchmark-tests.ts [hunk 3]
```diff
@@ -1,403 +1,640 @@
...
+import { MathOperator, Method, Payload } from '@joshdb/provider';
 import type { Benchmark } from '../Benchmark';
 
 export const BENCHMARK_TESTS: Benchmark.Test[] = [
   {
     name: Method.AutoKey,
 
     run: async ({ josh }) => {
-      await josh.autoKey();
+      await josh.autoKey({ method: Method.AutoKey, errors: [] });
```
Comment (RealShadowNova) [inline benchmarks/src/lib/constants/benchmark-tests.ts:9]:
```suggestion
      await josh[Method.AutoKey]({ method: Method.AutoKey, errors: [] });
```
This should be applied everywhere

File: benchmarks/src/lib/constants/benchmark-tests.ts [hunk 4]
```diff
@@ -1,403 +1,640 @@
...
   {
     name: Method.Dec,
 
     beforeAll: async ({ josh, entries }) => {
-      await josh.setMany(entries);
+      await josh.setMany({
+        method: Method.SetMany,
+        errors: [],
+        overwrite: true,
+        entries: entries.map(([key, value]) => {
```
Comment (RealShadowNova) [inline benchmarks/src/lib/constants/benchmark-tests.ts:38]: Does converting this hit perf? If so, consider changing this before the function runs?

**Review by RealShadowNova — CHANGES_REQUESTED** [review]

File: benchmarks/package.json (hunk 2 above)
Comment (dan-online) [reply benchmarks/package.json:10]: It was a bit, weird to do. I tried but it wouldn't output dist correctly

Comment (RealShadowNova) [reply benchmarks/package.json:10]: I see, we can look into that later then.

**Review by RealShadowNova — APPROVED** [review]

**Pull request merged**